### ISSUE TITLE
NCrystal_sample.comp compilation warnings

### ISSUE THREAD

**Issue by tkittel** [mctools/ncrystal#4]
McStas compilation of NCrystal_sample.comp gives warnings, we should see if they are possible to fix.

**Comment by tkittel** [comment]
```
NCrystal_sample.comp: In function ‘mcraytrace’:
NCrystal_sample.comp:229:5: warning: passing argument 3 of ‘ncrystal_crosssection’ from incompatible pointer type [enabled by default]
     ncrystal_crosssection(params.proc_scat,ekin,&dir,&xsect_scat);
     ^
In file included from NCrystal_sample.comp:71:0:
NCrystalLink/include/NCrystal/ncrystal.h:102:8: note: expected ‘const double (*)[3]’ but argument is of type ‘double (*)[3]’
   void ncrystal_crosssection( ncrystal_process_t,
        ^
NCrystal_sample.comp:269:7: warning: passing argument 3 of ‘ncrystal_genscatter’ from incompatible pointer type [enabled by default]
       ncrystal_genscatter( params.scat,ekin, &dir, &dirout, &delta_ekin);
       ^
In file included from NCrystal_sample.comp:71:0:
NCrystalLink/include/NCrystal/ncrystal.h:115:8: note: expected ‘const double (*)[3]’ but argument is of type ‘double (*)[3]’
   void ncrystal_genscatter( ncrystal_scatter_t,
        ^
NCrystal_sample.comp:295:9: warning: passing argument 3 of ‘ncrystal_crosssection’ from incompatible pointer type [enabled by default]
         ncrystal_crosssection(params.proc_scat,ekin,&dir,&xsect_scat);
         ^
In file included from NCrystal_sample.comp:71:0:
NCrystalLink/include/NCrystal/ncrystal.h:102:8: note: expected ‘const double (*)[3]’ but argument is of type ‘double (*)[3]’
   void ncrystal_crosssection( ncrystal_process_t,
        ^
```

**Comment by tkittel** [comment]
Interestingly enough, testing on SL6, gcc 4.4 does give similar warnings as seen on , but gcc5.4.0 does not.

**Comment by tkittel** [comment]
On OSX the warning goes away if e.g. the `ncrystal_crosssection` call gets a cast like:

```
    ncrystal_crosssection(params.proc_scat,ekin,(const double(*)[3])&dir,&xsect_scat);
```

That might be a short-term solution, but I am not sure I like such an interface. Perhaps we can find better signatures for ncrystal.h ? However, now is not a good time for interface changes...

**Comment by tkittel** [comment]
This is a real bug, not just a compiler feature: https://stackoverflow.com/questions/34488559/pointer-to-array-with-const-qualifier-in-c-c

Will add the `(const double(*)[3])` casts in NCrystal_sample.comp

**Comment by tkittel** [comment]
Fixed in NCrystal release v0.9.4.
